File: apps/macos/PasteSheets/Domain/UseCases/ManageItemsUseCase.swift
import Foundation

final class ManageItemsUseCase {
    private let repo: PasteItemRepository

    init(repo: PasteItemRepository) {
        self.repo = repo
    }

    func getAllItems() throws -> [PasteItem] {
        try repo.getAllItems()
    }

    func createItem(content: String, directory: String, memo: String?) throws -> Int64 {
        try repo.createItem(content: content, directory: directory, memo: memo)
    }

    func updateItem(id: Int64, content: String, directory: String, memo: String?) throws {
        try repo.updateItem(id: id, content: content, directory: directory, memo: memo)
    }

    func deleteItem(id: Int64) throws {
        try repo.deleteItem(id: id)
    }
}

File: apps/windows/PasteSheet/Domain/UseCases/ManageItemsUseCase.cs
using PasteSheet.Domain.Entities;
using PasteSheet.Domain.Repositories;

namespace PasteSheet.Domain.UseCases;

public sealed class ManageItemsUseCase
{
    private readonly IPasteItemRepository _repo;

    public ManageItemsUseCase(IPasteItemRepository repo) => _repo = repo;

    public List<PasteItem> GetAllItems() => _repo.GetAllItems();

    public long CreateItem(string content, string directory, string? memo) =>
        _repo.CreateItem(content, directory, memo);

    public void UpdateItem(long id, string content, string directory, string? memo) =>
        _repo.UpdateItem(id, content, directory, memo);

    public void DeleteItem(long id) => _repo.DeleteItem(id);
}
